src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java

```diff
@@ -0,0 +1,169 @@
+package hlf.java.rest.client.service.impl;
+
+import static hlf.java.rest.client.model.ChaincodeOperationsType.approve;
+import static java.util.Objects.isNull;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+import hlf.java.rest.client.exception.ErrorCode;
+import hlf.java.rest.client.exception.ServiceException;
+import hlf.java.rest.client.model.ChaincodeOperations;
+import hlf.java.rest.client.model.ChaincodeOperationsType;
+import hlf.java.rest.client.service.ChaincodeOperationsService;
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import lombok.extern.slf4j.Slf4j;
+import org.hyperledger.fabric.gateway.Gateway;
+import org.hyperledger.fabric.gateway.Network;
+import org.hyperledger.fabric.sdk.BlockEvent;
+import org.hyperledger.fabric.sdk.Channel;
+import org.hyperledger.fabric.sdk.HFClient;
+import org.hyperledger.fabric.sdk.LifecycleApproveChaincodeDefinitionForMyOrgProposalResponse;
+import org.hyperledger.fabric.sdk.LifecycleApproveChaincodeDefinitionForMyOrgRequest;
+import org.hyperledger.fabric.sdk.LifecycleCommitChaincodeDefinitionProposalResponse;
+import org.hyperledger.fabric.sdk.LifecycleCommitChaincodeDefinitionRequest;
+import org.hyperledger.fabric.sdk.Peer;
+import org.hyperledger.fabric.sdk.exception.InvalidArgumentException;
+import org.hyperledger.fabric.sdk.exception.ProposalException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class ChaincodeOperationsServiceImpl implements ChaincodeOperationsService {
+
+  @Autowired private Gateway gateway;
+
+  @Override
+  public String performChaincodeOperation(
+      String networkName,
+      ChaincodeOperations chaincodeOperationsModel,
+      ChaincodeOperationsType operationsType) {
+
+    validateChaincodeOperationsInput(chaincodeOperationsModel, operationsType);
+    HFClient hfClient = HFClient.createNewInstance();
+    Network network = gateway.getNetwork(networkName);
+    Channel channel = network.getChannel();
+
+    switch (operationsType) {
+      case approve:
+        {
+          return approveChaincode(hfClient, channel, chaincodeOperationsModel);
```

Comment:
Looks like a default instance of hfClient is created in this method and not initialized to any network or signer. It is then passed to the approve/commit.

Reply:
@arsulegai Done. thank you.